sdks/uniswapx-sdk/src/utils/OrderQuoter.ts

```diff
@@ -161,7 +171,7 @@ export class UniswapXOrderQuoter
   protected quoter: OrderQuoterContract;
 
   constructor(
-    protected provider: BaseProvider,
+    protected provider: JsonRpcProvider,
```

Comment:
why the change to JsonRpcProvider only?

Reply:
so with JsonRpcProvider we can send raw rpc calls which allows us to also optionally use the stateOverrides and blockOverrides fields